bridges/MastodonBridge.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+class MastodonBridge extends FeedExpander {
+
+	const MAINTAINER = 'husim0';
+	const NAME = 'Mastodon Bridge';
+	const CACHE_TIMEOUT = 900; // 15mn
+	const DESCRIPTION = 'Returns toots';
+	const URI = 'https://mastodon.social';
+
+	const PARAMETERS = [[
+		'canusername' => [
+			'name' => 'Canonical username (ex : @sebsauvage@framapiaf.org)',
+			'required' => true,
+		],
+		'norep' => [
+			'name' => 'Without replies',
+			'type' => 'checkbox',
+			'title' => 'Only return initial toots'
+		],
+		'noboost' => [
+			'name' => 'Without boosts',
+			'required' => false,
+			'type' => 'checkbox',
+			'title' => 'Hide boosts'
+		]
+	]];
+
+	public function getName(){
+		switch($this->queriedContext) {
+		case 'By username':
+			$param = 'canusername';
+			break;
+		default: return parent::getName();
+		}
+		
+		return $this->getInput($param);
+	}
+
+	protected function parseItem($newItem){
+		$item = parent::parseItem($newItem);
+
+		$content = str_get_html($item['content']);
+		$title = str_get_html($item['title']);
+
+		$item['title'] = substr($content->plaintext,0,75) . (strlen($content->plaintext) >= 75 ? '...' : '');
```

Comment:
Please take a look at `FacebookBridge` for a solution that takes whitespace into account.

https://github.com/RSS-Bridge/rss-bridge/blob/7926ffad7396fb029fe3c4c6d95f01e97921ab42/bridges/FacebookBridge.php#L660-L663

This should work in this case (not tested):

```suggestion
		$item['title'] = $content->plaintext;
		if(strlen($item['title']) > 75) {
			$item['title'] = substr($item['title'], 0, strpos(wordwrap($item['title'], 75), "\n")) . '...';
		}
```